pinecone/index_connection_test.go

```diff
@@ -22,93 +28,121 @@ import (
-	idxConnSourceTag *IndexConnection
-	vectorIds        []string
+	host              string
+	dimension         int32
+	apiKey            string
+	indexType         string
+	idxConn           *IndexConnection
+	sourceTag         string
+	idxConnSourceTag  *IndexConnection
+	vectorIds         []string
+	client            *Client
+	podIdxName        string
+	serverlessIdxName string
 }
 
 func TestIntegrationIndexConnection(t *testing.T) {
 	apiKey := os.Getenv("PINECONE_API_KEY")
 	assert.NotEmptyf(t, apiKey, "PINECONE_API_KEY env variable not set")
 
-	client, err := NewClient(NewClientParams{ApiKey: apiKey})
-	if err != nil {
-		t.FailNow()
-	}
+	client, err := NewClient(NewClientParams{ApiKey: apiKey, Headers: map[string]string{"content-type": "application/json"}})
+	require.NotNil(t, client, "Client should not be nil after creation")
+	require.NoError(t, err)
 
 	podIndexName := os.Getenv("TEST_PODS_INDEX_NAME")
 	assert.NotEmptyf(t, podIndexName, "TEST_PODS_INDEX_NAME env variable not set")
 
-	podIdx, err := client.DescribeIndex(context.Background(), podIndexName)
-	if err != nil {
-		t.FailNow()
-	}
-
-	podTestSuite := new(IndexConnectionTestsIntegration)
-	podTestSuite.indexType = "pod"
-	podTestSuite.host = podIdx.Host
-	podTestSuite.dimension = podIdx.Dimension
-	podTestSuite.apiKey = apiKey
-
 	serverlessIndexName := os.Getenv("TEST_SERVERLESS_INDEX_NAME")
 	assert.NotEmptyf(t, serverlessIndexName, "TEST_SERVERLESS_INDEX_NAME env variable not set")
 
-	serverlessIdx, err := client.DescribeIndex(context.Background(), serverlessIndexName)
-	if err != nil {
-		t.FailNow()
+	serverlessIdx := buildServerlessTestIndex(client, serverlessIndexName)
+	podIdx := buildPodTestIndex(client, podIndexName)
+
+	podTestSuite := &IndexConnectionTestsIntegration{
+		host:       podIdx.Host,
+		dimension:  podIdx.Dimension,
+		apiKey:     apiKey,
+		indexType:  "pods",
+		client:     client,
+		podIdxName: podIdx.Name,
 	}
 
-	serverlessTestSuite := new(IndexConnectionTestsIntegration)
-	serverlessTestSuite.indexType = "serverless"
-	serverlessTestSuite.host = serverlessIdx.Host
-	serverlessTestSuite.dimension = serverlessIdx.Dimension
-	serverlessTestSuite.apiKey = apiKey
+	serverlessTestSuite := &IndexConnectionTestsIntegration{
+		host:              serverlessIdx.Host,
+		dimension:         serverlessIdx.Dimension,
+		apiKey:            apiKey,
+		indexType:         "serverless",
+		client:            client,
+		serverlessIdxName: serverlessIdx.Name,
+	}
 
 	suite.Run(t, podTestSuite)
 	suite.Run(t, serverlessTestSuite)
 }
 
 func (ts *IndexConnectionTestsIntegration) SetupSuite() {
+	ctx := context.Background()
+
 	assert.NotEmptyf(ts.T(), ts.host, "HOST env variable not set")
 	assert.NotEmptyf(ts.T(), ts.apiKey, "API_KEY env variable not set")
-	additionalMetadata := map[string]string{"api-key": ts.apiKey}
+	additionalMetadata := map[string]string{"api-key": ts.apiKey, "content-type": "application/json"}
 
-	namespace, err := uuid.NewV7()
```

Comment:
Just changed this to make it more up to date w/current Go code I was seeing online